setup.py

```diff
@@ -7,25 +7,25 @@
 readme_path = Path(__file__).parent.joinpath("README.md")
 LONG_DESCRIPTION = readme_path.read_text()
 
-INSTALL_REQUIRES = ['aenum',
-                    'datamodel_code_generator[http]>=0.11.16',
-                    'paho-mqtt>=1.6.1',
-                    'pandas>=1.2',
-                    'pandas-datapackage-reader>=0.18.0',
-                    'pydantic>=2.0.2',
-                    'pydantic-settings>=2.0.1',
-                    'PyYAML',
+INSTALL_REQUIRES = ['aenum~=3.1.15',
+                    'datamodel_code_generator[http]~=0.25.0',
+                    'paho-mqtt~=1.6.1',
+                    'pandas~=1.3.5',
+                    'pandas_datapackage_reader~=0.18.0',
+                    'pydantic~=2.5.2',
+                    'pydantic-settings~=2.0.0',
                     'stringcase>=1.2.0',
-                    'igraph==0.9.8',
```

Comment:
dann ggf. direkt als optional. Genauso wie RDFLIB

Reply:
I add an optional module "semantics" in `setup.py`

Reply:
perfect. maybe also add rdflib to it ;)

Reply:
I add an instruction about the semantics module in readme. But I would first leave the rdflib as a required dependency because it is still coupled with some data models. This can be carefully removed in the future.

Reply:
perfect